safehttp/plugins/xsrf/xsrf.go

```diff
@@ -63,24 +56,55 @@ func Token(r *safehttp.IncomingRequest) (string, error) {
 	return tok.(string), nil
 }
 
+func addCookieID(w *safehttp.ResponseWriter) (*safehttp.Cookie, error) {
+	buf := make([]byte, 20)
+	_, err := rand.Read(buf)
+	if err != nil {
+		panic(fmt.Errorf("crypto/rand.Read: %v", err))
```

Comment:
I am nervous about this `panic` in the request path. Can this return the error instead?

Reply:
I changed it to return an error instead, but shouldn't a system failure result in a panic?